### PR TITLE
TP: adds the ability to rearrange server table columns as well as toggle column visibility/searchability

### PR DESCRIPTION
**Pull request by mitchell852**
## What does this PR (Pull Request) do?

- [x] This PR is not related to any Issue

This PR adds checkboxes to toggle the visiblity/searchability of server table columns as well as adds the ability to reorder the columns.

image

## Which Traffic Control components are affected by this PR?

- Traffic Portal

## What is the best way to verify this PR?
1. Run the [UI tests](https://github.com/apache/trafficcontrol/tree/master/traffic_portal/test/end_to_end)
2. Navigate to https://tp.doman.com/#!/servers and

1. reorder the columns; refresh the screen to ensure that the new order persists itself across browser sessions.
2. click on the drop down to toggle the visiblity/searchability of columns; refresh the screen to ensure that the selected columns are visible/hidden across browser sessions. also, ensure that values in hidden columns are not considered in the table search.

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/4045/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/4047/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/4061/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/4064/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/4086/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/4088/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/4090/
Test PASSed.

File: traffic_portal/test/end_to_end/Servers/servers-spec.js [hunk 1]
```diff
@@ -69,6 +69,18 @@ describe('Traffic Portal Servers Test Suite', function() {
 		expect(browser.getCurrentUrl().then(commonFunctions.urlPath)).toEqual(commonFunctions.urlPath(browser.baseUrl)+"#!/servers");
 	});
 
+	it('should toggle the visibility of the first table column ', function() {
+		browser.driver.findElement(by.id('toggleColumns')).click();
+		let first = element.all(by.css('input[type=checkbox]')).first();
+		expect(first.isSelected()).toBe(true);
+		browser.sleep(2000);
+		first.click();
+		browser.sleep(2000);
```
Comment (ocket8888) [inline traffic_portal/test/end_to_end/Servers/servers-spec.js]: are these `sleep`s supposed to be here?

File: docs/source/admin/traffic_portal/usingtrafficportal.rst [hunk 2]
```diff
@@ -296,26 +296,55 @@ Interfaces for managing the various components of Traffic Control and how they i
...
+:Network IP:		[Visible by default] The IPv4 address of ``interfaceName``
+:Network MTU:		The Maximum Transmission Unit (MTU) to configured on ``interfaceName``
+:Network Subnet:	The IPv4 subnet mask used by ``interfaceName``
+:Offline Reason:	A user-entered reason why the server is in ADMIN_DOWN or OFFLINE status
+:Phys Location:		[Visible by default] The name of the physical location where the server resides
+:Profile:			[Visible by default] The :ref:`profile-name` of the :term:`Profile` used by this server
+:Rack:				A string indicating "server rack" location
+:Reval Pending:		[Visible by default] A boolean value represented as a clock (content invalidation/revalidation is pending) or green check mark (content invalidation/revalidation is not pending)
+:Router Hostname:	The human-readable name of the router responsible for reaching this server
+:Router Port Name:	The human-readable name of the port used by the router responsible for reaching this server
```
Comment (ocket8888) [inline docs/source/admin/traffic_portal/usingtrafficportal.rst]: TP doesn't allow you to set/change router hostname/port name, and honestly I have no idea what those could be. TR hostname and port... name? TO clearly doesn't actually populate that automatically, so you'd need to set it yourself and remember to update it manually on infrastructural changes.

So I guess my question is: does this need to be a column?

File: traffic_portal/test/end_to_end/Servers/servers-spec.js (hunk 1 above)
Comment (mitchell852) [reply traffic_portal/test/end_to_end/Servers/servers-spec.js]: no. gripes. will fix.

File: docs/source/admin/traffic_portal/usingtrafficportal.rst (hunk 2 above)
Comment (mitchell852) [reply docs/source/admin/traffic_portal/usingtrafficportal.rst]: well, it is a property on the server that somebody may or may not want to see so i'd rather leave it up to the user.

Comment (mitchell852) [reply docs/source/admin/traffic_portal/usingtrafficportal.rst]: hmm. i'm surprised TP doesn't allow you...actually, maybe you are right. i think i will drop them.

Comment (mitchell852) [reply docs/source/admin/traffic_portal/usingtrafficportal.rst]: you can define those values in TP on the server form

image

so i think it's a good idea to have them accessible in the servers table.

**Comment by mitchell852** [comment]
@ocket8888 - changes made as requested

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/4103/
Test PASSed.

**Comment by mitchell852** [comment]
fixes #2385

**Pull request merged**